### PR TITLE
[WIP] voodoo/metagen command

### PR DESCRIPTION
**Pull request by manasdk**
miserable attempt at integrating metagen 

(lol .. named branch meatgen)

### CODE REVIEW

**Comment by manasdk** [comment]
kami - maybe there is a better way to do this but I was unable to get positional and optional args to mix with cmd2. Any idea?

**Comment by manasdk** [comment]
The only alternative I can come up with is to completely ignore the @options and apply argparse on the input to the do_metagen method.

**Comment by manasdk** [comment]
In 39db472adf839bba3717723e9bf4e752fb591f4c I am using argparse. Not perfect but at least takes away a ton of pain. Makes it quite tolerable.

**Comment by Kami** [comment]
@manasdk For optional arguments you need to use `@options` decorator with cmd2.

I will look at the other changes shortly.

File: st2sdk/metagen/cmdmixin.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+COMMAND_HELP = {
+    'metagen': [
+        'metagen [pack name] [module name]',
+        'Create initial directory structure for the provided pack. Inspect module and autogenerate'
+        'action metadata.'
+    ]
+}
+
+
+class Data(object):
```
Comment (Kami) [inline st2sdk/metagen/cmdmixin.py:8]: I find this class / abstraction slightly confusing - what benefits does it provide over a dict?

Comment (manasdk) [inline st2sdk/metagen/cmdmixin.py:8]: Its temp. Playing around with some options without actually changing metagen. I put it in place as metagen expects an object.

**Comment by manasdk** [comment]
kami : `@options` is pretty horrible to work with. Pretty much need to duplicate help and type info. I worked around with argparse.

File: st2sdk/metagen/cmdmixin.py (hunk 1 above)
Comment (Kami) [inline st2sdk/metagen/cmdmixin.py:8]: Ah, OK.

Overall, it looks good to me. I'm fine with merging it as soon as the build is fixed.

**Comment by Kami** [comment]
I'm fine with using argparse as long as we use it consistency across the code base.

I know this is just a quick port so I'm fine with not refactoring any code right now, but eventually, we should build some more common building blocks so maintaining this tool and adding new commands will be easy and painless.

**Comment by manasdk** [comment]
> I'm fine with using argparse as long as we use it consistency across the code base.

I was too optimistic about argparse and using it the way I did; it doesn't actually work. Not sure what the heck happened.

**Comment by Kami** [comment]
Did you figure out the argparse stuff?

If not, I think using`sys.argv` here `data = self._get_args_parser_metagen().parse_args(args)` instead of `args` might work.

**Comment by manasdk** [comment]
#fail

**Pull request closed without merging**